azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -80,6 +80,12 @@ class MQTTTransport(object):
     :type on_mqtt_connection_failure_handler: Function
     """
 
+    # Horton uses this variable to make sure we're cleaning up correctly.
+    # This object in particuar is checked because it holds a reference to Paho
+    # and Paho holds references to network resources and we want to be extra sure
+    # that we clean those up correctly.
+    _object_count = 0
```

Comment:
I don't quite understand the purpose, since the only thing done with this value is increment and decrement. There seem to be no implementation changes based on the value, so doesn't that mean that nothing has changed?

Reply:
Yes, Horton uses (or will use) this variable to make sure we're cleaning up correctly.  The Python glue checks this as part of test teardown to make sure no objects were leaked.  It's kludgy but I need to know I'm on stable ground for my dropped connection tests. 

---
In reply to: [329764771](https://github.com/Azure/azure-iot-sdk-python/pull/334#discussion_r329764771) [](ancestors = 329764771)